lib/api/controller/security.js

```diff
@@ -1159,13 +1165,25 @@ class SecurityController extends NativeController {
    * @returns {Promise}
    * @private
    */
-  async _persistUser(request, profileIds, content) {
-    const id = this.getId(request, ifMissingEnum.GENERATE);
+  async _persistUser(request, profileIds, content, humanReadableId=true) {
+    const generator = humanReadableId ? nameGenerator : uuidv4;
     const credentials = this.getBodyObject(request, 'credentials', {});
     const strategies = Object.keys(credentials);
-
+    
+    let id = null;
```

Comment:
Nitpicking: if you want to initialize this variable then you could use an empty string to keep the same type after assignment